pkg/service/restore/worker.go

```diff
@@ -161,12 +161,96 @@ func (w *worker) initTarget(ctx context.Context, properties json.RawMessage) err
 		} else {
 			w.logger.Info(ctx, "Found schema file")
 		}
+		return nil
+	}
+
+	// The only way to have a list of dcs from the source cluster is to
+	// get them from backup locations.
+	sourceDC, err := w.collectDCFromLocation(ctx, t.locationHosts)
```

Comment:
Correct, but we are already going through all of the manifests in the `GetTarget` (`initUnits`), so we should probably get all of the interesting data within a single iteration.